src/obj_dir.ml

```diff
@@ -148,15 +154,26 @@ module Local = struct
     match cm_kind with
     | Cm_kind.Cmx -> native_dir t
     | Cmo | Cmi -> byte_dir t
+
+  let cm_public_dir t (cm_kind : Cm_kind.t) =
+    match cm_kind with
+    | Cmx -> native_dir t
+    | Cmo -> byte_dir t
+    | Cmi -> public_cmi_dir t
 end
 
-type t =
-  | External of External.t
-  | Local of Local.t
+type _ t =
+  | External      : External.t -> Path.t t
+  | Local         : Local.t    -> Path.Build.t t
```

Comment:
The `Local` constructor is only used in `make_lib`, which is unused in the code base

Reply:
I mean that it is the only place it is used in an expression to construct a value. If we remove `make_lib`, then we can remove `Local`. And if we remove `Local`, then all values are of type `Path.t t`

Reply:
Yeah, that is because we still have the `Obj_dir.Local` functions. If we adopt this PR, then we will not need most of the functions in `Object.Local.t` anymore and the local constructor will be more useful.

Just to clarify, this PR is still WIP in this sense. If it is accepted, I will remove `Object.Local` from the interface (or perhaps leave it `Obj_dir.Local.t` as an alias for `Path.Build.t Obj_dir.t`

Reply:
Ok, thanks for the info